diff --git a/src/XMonad/Core.hs b/src/XMonad/Core.hs
index 138d735..8d20bb5 100644
--- a/src/XMonad/Core.hs
+++ b/src/XMonad/Core.hs
@@ -627,6 +627,7 @@ recompile force = io $ do
                           , "-fforce-recomp"
                           , "-main-is", "main"
                           , "-v0"
+                          , "-O2"
                           , "-o", bin
                           ] (Just dir) Nothing Nothing Nothing (Just errHandle)
        compileScript bin dir script errHandle =
